### ISSUE TITLE
Documenting the Tiki 100 400k format

### ISSUE THREAD

**Issue by tingox** [davidgiven/fluxengine#225]
[Tiki 100](https://en.wikipedia.org/wiki/Tiki_100) was a Norwegian computer running Tiko, a CP/M 2.2 compatible operating system. It used 5.25 inch floppies, 40 track or (later 80 track). This is a dump of a 40 track floppy, read on a 80 track drive, using the following command:
```
tingo@z30b:~/personal/projects/psoc/fluxengine/tmp_tiki$ ../fluxengine read ibm -o 525-tiki_100_400k_tiko_4.01_tikos_2.00_25.img -f 525-tiki_100_400k_tiko_4.01_tikos_2.00_25.flux
[..]
Autodetecting output geometry
H.SS Tracks --->
0. 0 ........................................
0. 1 ........................................
0. 2 ........................................
0. 3 ........................................
0. 4 ........................................
0. 5 ........................................
0. 6 ........................................
0. 7 ........................................
0. 8 ........................................
0. 9 ........................................
1. 0 ........................................
1. 1 ........................................
1. 2 ........................................
1. 3 ........................................
1. 4 ........................................
1. 5 ........................................
1. 6 ........................................
1. 7 ........................................
1. 8 ........................................
1. 9 ........................................
Good sectors: 800/800 (100%)
Missing sectors: 0/800 (0%)
Bad sectors: 0/800 (0%)
writing 40 tracks, 2 heads, 10 sectors, 512 bytes per sector, 400 kB total
```
As far as I know, there is nothing confidential on this image. Using the correct definition in diskdefs (cpmtools), cpmls can read the directory:
```
tingo@z30b:~/personal/projects/psoc/fluxengine/tmp_tiki$ cpmls -f tik3 -l 525-tiki_100_400k_tiko_4.01_tikos_2.00_25.img
0:
-rwxrwxrwx     256 Jan 01 1970  !.com
-rwxrwxrwx    3840 Jan 01 1970  120.com
-rwxrwxrwx    3840 Jan 01 1970  160.com
-rwxrwxrwx    2816 Jan 01 1970  160x.com
-rwxrwxrwx    5504 Jan 01 1970  40.com
-rwxrwxrwx    5760 Jan 01 1970  40x12.com
-rwxrwxrwx    4352 Jan 01 1970  72i.com
-rwxrwxrwx    5504 Jan 01 1970  80.com
-rwxrwxrwx    4096 Jan 01 1970  8088.com
-rwxrwxrwx    5376 Jan 01 1970  80adm.com
-rwxrwxrwx    5376 Jan 01 1970  80inf.com
-rwxrwxrwx    5376 Jan 01 1970  80mon.com
-rwxrwxrwx    2816 Jan 01 1970  80x.com
-rw-rw-rw-     128 Jan 01 1970  address.dta
-rw-rw-rw-    3072 Jan 01 1970  animal.bbb
-rw-rw-rw-     128 Jan 01 1970  animal.dat
-rw-rw-rw-     384 Jan 01 1970  asembly.bbb
-rwxrwxrwx   16512 Jan 01 1970  basic.com
-rw-rw-rw-    1920 Jan 01 1970  basic.txt
-rw-rw-rw-     512 Jan 01 1970  dyr.bbb
-rw-rw-rw-     896 Jan 01 1970  dyr.fnt
-rwxrwxrwx    7552 Jan 01 1970  edit.com
-rw-rw-rw-     768 Jan 01 1970  envelope.bbb
-rw-rw-rw-    1792 Jan 01 1970  farger.bbb
-rw-rw-rw-    7168 Jan 01 1970  findex.bbb
-rw-rw-rw-     640 Jan 01 1970  flate2.bbb
-rw-rw-rw-    1408 Jan 01 1970  frand0.bbb
-rw-rw-rw-    2304 Jan 01 1970  frand1.bbb
-rw-rw-rw-    8064 Jan 01 1970  frand2.bbb
-rwxrwxrwx     512 Jan 01 1970  ibm.com
-rw-rw-rw-     384 Jan 01 1970  icos2.bbb
-rw-rw-rw-    1664 Jan 01 1970  info.tex
-rwxrwxrwx   11392 Jan 01 1970  install.com
-rw-rw-rw-     384 Jan 01 1970  isinc.bbb
-rwxrwxrwx    4096 Jan 01 1970  kat.com
-rwxrwxrwx    4224 Jan 01 1970  kopi.com
-rw-rw-rw-    7680 Jan 01 1970  les.meg
-rwxrwxrwx     256 Jan 01 1970  lesmeg.com
-rwxrwxrwx    3072 Jan 01 1970  meny.com
-rw-rw-rw-    2048 Jan 01 1970  meny.txt
-rw-rw-rw-     768 Jan 01 1970  merge.bbb
-rw-rw-rw-     128 Jan 01 1970  name.ndx
-rwxrwxrwx    3840 Jan 01 1970  nyplate.com
-rwxrwxrwx   15488 Jan 01 1970  palett.com
-rw-rw-rw-     256 Jan 01 1970  palett.dat
-rw-rw-rw-     768 Jan 01 1970  piano.bbb
-rw-rw-rw-    2048 Jan 01 1970  plater.bak
-rw-rw-rw-    2304 Jan 01 1970  plater.std
-rwxrwxrwx    8960 Jan 01 1970  pyton.com
-rwxrwxrwx    8960 Jan 01 1970  r]kopi.com
-rwxrwxrwx   29568 Jan 01 1970  robot.com
-rw-rw-rw-   23296 Jan 01 1970  robot.doc
-rw-rw-rw-    1152 Jan 01 1970  rothus.bbb
-rwxrwxrwx     384 Jan 01 1970  scandis.com
-rwxrwxrwx    2048 Jan 01 1970  sett.com
-rw-rw-rw-     384 Jan 01 1970  sinc.bbb
-rw-rw-rw-     384 Jan 01 1970  sinc2.bbb
-rw-rw-rw-     128 Jan 01 1970  sinus.bbb
-rw-rw-rw-     384 Jan 01 1970  sirkel.bbb
-rwxrwxrwx   24576 Jan 01 1970  sjakk.com
-rw-rw-rw-     640 Jan 01 1970  slager.bbb
-rw-rw-rw-     256 Jan 01 1970  sol.bbb
-rwxrwxrwx    2432 Jan 01 1970  tdv.com
-rwxrwxrwx    2816 Jan 01 1970  tdvdef.com
-rw-rw-rw-    8960 Jan 01 1970  tiko200b.sys
-rw-rw-rw-    8960 Jan 01 1970  tikoab.sys
-rwxrwxrwx    2048 Jan 01 1970  tikokopi.com
-rw-rw-rw-   12288 Jan 01 1970  tikos.sys
-rwxrwxrwx    1280 Jan 01 1970  unlist.com
-rw-rw-rw-    1408 Jan 01 1970  ur.bbb
-rwxrwxrwx    1792 Jan 01 1970  utf\r.com
-rw-rw-rw-    1536 Jan 01 1970  viktig.txt
-rw-rw-rw-    1664 Jan 01 1970  xdir.cmd
```

[525-tiki_100_400k_tiko_4.01_tikos_2.00_25.zip](https://github.com/davidgiven/fluxengine/files/5888259/525-tiki_100_400k_tiko_4.01_tikos_2.00_25.zip)

**Comment by tingox** [comment]
Here are the cpmtools definitions for the Tiki 100 formats:
```
# TIK1 TIKI-100 90K - SSSD 48 tpi 5.25" - 128 x 18
diskdef tik1
seclen 128
tracks 40
sectrk 18
blocksize 1024
maxdir 32
skew 5
boottrk 3
os 2.2
end

# TIK2 TIKI-100 200K - SSDD 48 tpi 5.25" - 512 x 10
diskdef tik2
seclen 512
tracks 40
sectrk 10
blocksize 1024
maxdir 64
skew 1
boottrk 2
os 2.2
end

# TIK3 TIKI-100 400K - DSDD 48 tpi 5.25" - 512 x 10
diskdef tik3
seclen 512
tracks 80
sectrk 10
blocksize 2048
maxdir 128
skew 1
boottrk 2
os 2.2
end

# TIK4 TIKI-100 800K - DSDD 96 tpi 5.25" - 512 x 10
diskdef tik4
seclen 512
tracks 160
sectrk 10
blocksize 2048
maxdir 128
skew 1
boottrk 2
os 2.2
```
Hope it helps.

**Comment by davidgiven** [comment]
Very very belatedly... I have just added profiles for the four Tiki formats (called tiki90, tiki200, tiki400 and tiki800). I've tested the CP/M filesystem support for the 200kB and 400kB formats and just kinda guessed for the other two --- if you're still interested in this, let me know if they work? Thanks very much!

They're currently read-only. They can be made read-write but that'll need a real machine to test the written disks with.
